plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintCheck.kt

```diff
@@ -43,6 +46,11 @@ open class KtlintCheck @Inject constructor(objectFactory: ObjectFactory) : Defau
         }
     }
 
+    @get:SkipWhenEmpty
+    @get:PathSensitive(PathSensitivity.RELATIVE)
+    @get:InputFiles
+    val editorConfigFiles: FileCollection = project.rootProject.files(EDITORCONGIG_FILE_NAME)
+        .plus(project.files(EDITORCONGIG_FILE_NAME))
```

Comment:
Do you want to use a `RegularFileProperty` instead? Then you can annotate it as `@get:InputFile` instead of `@get:InputFiles`. Unless you think the user can specify multiple files.

Also, do you want to consider making this field `internal` or do you want to allow users to modify it?

Reply:
Totally makes sense.

Another problem with the way you have this is that the root project is not always necessarily in a parent directory.
It is completely valid to have a structure like this:
```
someDir
 - rootProject
   - rootProject.gradle
   - .editorconfig
   - settings.gradle
 - subprojects
   - subprojects.gradle
   - .editorconfig
     - subprojectA
       - subprjoectA.gradle
       - .editorconfig
```
I understand this is an atypical project structure. But it is completely valid.

See the "Where are these files stored?" section from this document to see how the filesystem walking should really happen:
https://editorconfig.org/

Reply:
Ok, I will try to adapt proposed strategy for detecting `.editorconfig` files that is seems to be similar how it is in the Ktlint :+1: